app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -20,7 +20,11 @@ def interstitial
       if service.candidate_does_not_have_a_course_from_find || service.candidate_has_submitted_application
         if more_reference_needed? && FeatureFlag.active?('show_new_referee_needed')
           redirect_to candidate_interface_additional_referee_path
+        elsif flash[:success] == t('apply_from_find.account_created_message') && FeatureFlag.active?('before_you_start')
+          flash[:success] = t('apply_from_find.account_created_message')
```

Comment:
It seems that checking for the flash messages presence stops it beings passed on to the next controller action. Is there any way to retain the value and check for it?

Reply:
There is also `#keep` https://apidock.com/rails/ActionController/Flash/FlashHash/keep

Reply:
Thanks for that. That's really helpful! I think Theo's right that I shouldn't be checking the flash, but that's really useful to know going forward.